presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ApiRequester.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller;
+
+import org.eclipse.jetty.http.HttpMethod;
+import org.glassfish.jersey.client.JerseyClient;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import java.net.URI;
+import java.util.concurrent.Future;
+
+import static java.util.Objects.requireNonNull;
+import static org.glassfish.jersey.client.JerseyClientBuilder.createClient;
+
+public class ApiRequester
+{
+    private UriBuilder uriBuilder;
+    private HttpMethod method;
+    private Entity entity;
+    private MultivaluedMap<String, Object> headers;
+    private String mediaType;
+
+    private static JerseyClient jerseyClient = createClient();
```

Comment:
I think that if we figure out how to inject it, we have at least two options to close it.
1. Figure out Guice scopes, and create a scope that closes it for us.
2. Initialize it outside in a `try`-with-resources, and inject the instance as a singleton.